src/afrolabs/components/http.clj

```diff
@@ -61,8 +73,16 @@
                        (map #(partial handle-http-request
                                       %)
                             handlers))
-        handler (ring.middleware.pratchett/wrap-pratchett handler)
-        s (httpkit/run-server handler
+        middleware-chain (or (when (seq middleware-providers)
+                               (->> middleware-providers
+                                    (mapcat get-middleware)
+                                    (reverse)
+                                    (apply comp)))
+                             identity)
+        handler' (-> handler
+                     (middleware-chain)
+                     (ring.middleware.pratchett/wrap-pratchett))
```

Comment:
Setting up the middleware chain.
- if now middleware provided, just use `identity` which is easier than nil-checking everywhere
- apply the middleware outside-in, so that middleware supplied "first", will see the request first and the response last, whereas middleware provided "last", will see the request last, and the response first.
- compose with `ring.middleware.pratchett` which is applied to everything, everywhere, all the time.